### ISSUE TITLE
How to flash rubber duck?

### ISSUE THREAD

**Issue by zutiwe** [hak5darren/USB-Rubber-Ducky#96]
https://github.com/hak5darren/USB-Rubber-Ducky/wiki/Flashing-ducky#DuckyFlasher_Utility

I'm trying to flash my brand new duck. The` ducky-flasher` recognizes the duck is in dfu mode and seems to be working:

>       Welcome to ducky-flasher
>               _.._            
>              /   a\__,        
>              \  -.___/        
>               \  \            
>          (\____)  \           
>      |\_(         ))          
> _____|   (_        /________  
>      _\____(______/__         
>           ______              
> 
> 
> Which firmware would you like to flash?
> 1. Original (duck.hex)
> 2. FAT Duck (usb.hex)
> 3. Detour Duck (m_duck.hex)
> 4. Twin Duck (c_duck.hex)
> 
> 99. Exit ducky-flasher
> 
> Choice: 4
> [3;J
> 
> 
> Description of Twin Duck Firmware
> 
> There are 3 versions of the Twin Duck Firmware
> Original, Special Request 1, and Special Request 2!
> 
> The original executes on inserting it
> Special Request 1 executes on LED trigger (Cap Locks, Num Locks, etc)
> Special Request 2 executes on pressing the GPIO pin
> 
> Which version do you want to flash?
> 1 = Original
> 2 = Special Request 1
> 3 = Special Request 2
> Choice: 1
> Validating...
> 34908 bytes used (13.75%)
> [3;J
> 
> 
> [3;J
> 
> Your duck has been flashed, would you like to return to the main menu?
> 1 = yes
> 2 = no
> Choice: 2
> [3;J
> 
> 

After, the duck continuously flashes red and green non-stop. Only one 122 MB partition shows up in my file manager.

Maybe I have the concept of "twinduck" all wrong. I thought it separates the  microSD into partitions, one partition for inject.bin and the other (larger) partition for exfiltrating data locally?

- What does the red/green flashing mean?
- Is there a twinduck*.hex?
- How the hell do I enable twinduck mode??

**Comment by Siegfried5** [comment]
Ok so here are some basic trouble shooting thing. 



First did you remove the micro SD card from your duck.  If so did you install dfu-programmer on your machine.

**Comment by zutiwe** [comment]
Yeah, I installed the`dfu-programmer` in Xenial. I followed all the steps in the above linked wiki. I tried both without and with the microSD (before creating this git issue) and no real differences that I noticed. The light just blinks red/green over and over after the `ducky-flasher` completes.

**Comment by zutiwe** [comment]
My solution:

>sudo dfu-programmer at32uc3b1256 erase
>sudo dfu-programmer at32uc3b1256 flash --suppress-bootloader-mem c_duck_v2.1.hex
>sudo dfu-programmer at32uc3b1256 reset

https://github.com/hak5darren/USB-Rubber-Ducky/blob/master/Firmware/Images/c_duck_v2.1.hex

IMPORTANT! When saving `c_duck_v2.1.hex`, do it manually. Using wget or curl or git it didn't work for me. I copy/pasted the .hex text into a file and saved it locally. This for some reason seemed to work. Flashing completes without error and I can now access the mass storage in twinduck mode.

**Comment by mfaerevaag** [comment]
I'm having the same problem. Bought three rubber duckies, the first flashed with no problems. The two others doesn't work with the `c_duck_v2.1.hex` firmware.

Have also tried with and without SD card. No problems with the `c_duck_v2.1.hex` file.

When using the "Duck Programming Script" (`program.bat`) I get the following error message:
`WARNING: The user program and the bootloader overlap!`.

dfu-programmer reports no errors.

**Comment by rootkali62** [comment]
Hello you guys, i am having the same issue here. here is some info, I am currently on windows 10 and flashed my duck to twinduck and it seemed to be ok but nothing shows up on my devices... it only shoes in "Device manager" as "portable device" nothing else. when i tried to run a script on my self nothing works anymore... is there anyway i can just remove twinduck? or can someone help me out here please!
